perlock-core/src/main/java/name/mitterdorfer/perlock/SinglePathWatcherFactory.java

```diff
@@ -0,0 +1,107 @@
+package name.mitterdorfer.perlock;
+
+import name.mitterdorfer.perlock.impl.util.Preconditions;
+
+import java.nio.file.Path;
+
+import java.nio.file.WatchEvent;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static java.nio.file.StandardWatchEventKinds.*;
+
+/**
+ * Factory to create {@link PathWatcher}s that only deal with a single file and
+ * ignore change to other files within the directory
+ */
+public class SinglePathWatcherFactory {
```

Comment:
I can move these into the `PathWatcherFactory`.
In regards to checking that a `Path` represents a file, do you mean checking if `Files#isRegularFile`?
I am not sure that check is necessary or even desirable, it is conceiveable that the user may want to monitor the creation and deletion of a particular directory. So long as the path is non-null I think it would suffice.